### PR TITLE
Abstract implementation for iScheduledProcess

### PR DESCRIPTION
**Pull request by piRGoif**
In iTop, BackupExec is a iScheduledProcess implementation that contains some code (getting config parameters and next occurrence datetime) that could be made generic and reusable by extensions.

This PR moves this code to a new AbstractScheduledProcess class that can be used by other modules.

### CODE REVIEW

File: datamodels/2.x/itop-backup/main.itop-backup.php [hunk 1]
```diff
@@ -85,16 +85,26 @@ public function ListFiles($sBackupDir)
 	}
 }
 
-class BackupExec implements iScheduledProcess
+class BackupExecProcess extends AbstractScheduledProcess
```
Comment (bruno-ds) [inline datamodels/2.x/itop-backup/main.itop-backup.php]: Why did you rename it?
Could it cause BC issues?

File: core/backgroundprocess.inc.php [hunk 2]
```diff
@@ -67,14 +68,180 @@ interface iScheduledProcess extends iProcess
 	public function GetNextOccurrence();
 }
 
+
 /**
- * Class ProcessException
- * Exception for iProcess implementations.<br>
+ * Implementation of {@link iScheduledProcess}, using config parameters for module
```
Comment (bruno-ds) [inline core/backgroundprocess.inc.php]: we loose the git history, couldn't it have been possible to preserve it?

File: datamodels/2.x/itop-backup/main.itop-backup.php (hunk 1 above)
Comment (piRGoif) [reply datamodels/2.x/itop-backup/main.itop-backup.php]: There will at least be an issue with existing records in the BackgroundTask table... (no bugs but existing records kept)
But as my goal was to discuss with the rest of the team, I made this proposal also :)

File: core/backgroundprocess.inc.php (hunk 2 above)
Comment (piRGoif) [reply core/backgroundprocess.inc.php]: Discussed this with Bruno : he was worried about the blame that we are loosing on the 2 methods that were moved from BackupExec to the new AbstractScheduledProcess class.
But indeed those 2 methods are quite old, both are dated from 2 months in 2014, and only 1 line is more recent, but has a comment identical to the commit message.
Bruno and I agree that loosing this history won't be a problem.

File: datamodels/2.x/itop-backup/main.itop-backup.php (hunk 1 above)
Comment (piRGoif) [reply datamodels/2.x/itop-backup/main.itop-backup.php]: The dev team did a review today.
Decisions :
* rollback the BackupExec class renaming (some risk introduced with a very slight benefit)
* important to have this in 2.7.0 as it will allow to change corresponding extensions sooner
* apart of the renaming rollback ok to merge

I'm closing this conversation, many thanks again Bruno for the feedback !

**Review by Molkobain — DISMISSED** [review]

**Review by Molkobain — APPROVED** [review]

**Review by bruno-ds — APPROVED** [review]
I had troubble understanding the mean of $aWEEKDAYTON 
aka : WEEKDAY TO N 

but seems all good to me :)
